gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/KubernetesUndeployTask.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.gradle.plugin.task;
+
+import org.eclipse.jkube.gradle.plugin.KubernetesExtension;
+import org.eclipse.jkube.kit.common.util.ResourceUtil;
+import org.eclipse.jkube.kit.config.resource.ResourceConfig;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.io.IOException;
+import java.util.Optional;
+
+public class KubernetesUndeployTask extends AbstractJKubeTask {
+
+  @Inject
+  public KubernetesUndeployTask(Class<? extends KubernetesExtension> extensionClass) {
+    super(extensionClass);
+    setDescription("Undeploys (deletes) the kubernetes resources generated by the current project.");
+  }
+
+  @Override
+  public void run() {
+    try {
+      undeploy();
```

Comment:
What's the purpose of this single method call here? why not just add the body of the undeploy method? Is this supposed to be extended elsewhere (OpenShift)? Might it make more sense to create an extensible findResources method instead?